### PR TITLE
fix: directly return an error message when there is no healthy node available

### LINKED ISSUE #111: Bug: when no endpoints is healthy, "it will cause crazy retries"
#109 fixed the situation when some endpoints got taints it could be jumped over. But when all endpoints are unhealthy, it keeps the old method, so the problem of #101 still not fixed and "it will cause crazy retries". See [discussion](https://github.com/api7/lua-resty-etcd/pull/109#discussion_r560273225)

### PR DESCRIPTION
**Pull request by tzssangglass**
fix #111

### CODE REVIEW

**Comment by tzssangglass** [comment]
@Yiyiyimu @membphis @nic-chen pls take a look

**Review by membphis — APPROVED** [review]

File: lib/resty/etcd/v3.lua [hunk 1]
```diff
@@ -211,7 +211,7 @@ local function choose_endpoint(self)
                 return endpoint
             end
         end
-        utils.log_warn("has no healthy endpoint")
```
Comment (nic-chen) [inline lib/resty/etcd/v3.lua:214]: we could retain the log here

**Comment by tzssangglass** [comment]
update, retain the log

**Comment by membphis** [comment]
merged, many thx @tzssangglass

**Pull request merged**